### PR TITLE
matchingPieces-base

### PR DESCRIPTION
**Pull request by RafalZmija**

### CODE REVIEW

File: static/js/Game.js [hunk 1]
```diff
@@ -114,11 +117,10 @@ class Game {
 
                 if (floor[i + 4][j + 7] == 1) {
                     let playerID = 3
-                    console.log(j * i + i)
                     let imageOnTopName = startImages()[imageCounterStrike]
                     let topMaterialPath = `./gfx/${imageOnTopName}.png`;
                     let pieceID = imageOnTopName
-                    console.log("TOP MATERIAL PATH: " + topMaterialPath)
+                    //console.log("TOP MATERIAL PATH: " + topMaterialPath)
```
Comment (n2oneProgrammer) [inline static/js/Game.js:123]: just remove it not comment

File: static/js/Game.js [hunk 2]
```diff
@@ -114,11 +117,10 @@ class Game {
...
                     const piece = new Piece(playerID, pieceID, topMaterialPath);
                     // console.log(piece);
```
Comment (n2oneProgrammer) [inline static/js/Game.js:125]: useless comment

File: static/js/Game.js [hunk 3]
```diff
@@ -217,21 +219,45 @@ class Game {
         }
     }
     clickPieces = () => {
+        if (lastClickedPieceImageObj != undefined)
+            console.log("ostatnio kliknięty: " + lastClickedPieceImageObj.pieceID)
         this.raycaster.setFromCamera(this.pointer, this.camera);
         const intersects = this.raycaster.intersectObjects(this.scene.children);
         // console.log(intersects.length)
```
Comment (n2oneProgrammer) [inline static/js/Game.js:226]: useless

File: static/js/Game.js [hunk 4]
```diff
@@ -217,21 +219,45 @@ class Game {
...
-                    opacity: 1,
-                });
-                //console.log(intersects[0].object);
+        console.log("teraz clicknięty: " + intersects[0].object.pieceID);
+
+        if (lastClickedPieceImageObj != undefined) {
+            //console.log(intersects[0].object)
+            for (let i = 0; i < intersects.length; i++) {
+                if (intersects.length > 0 && intersects[0].object.name == "clickable" && lastClickedPieceImageObj.pieceID == intersects[0].object.pieceID) { // && lastClickedPieceImageObj.pieceID == intersects[0].object.pieceID
+                    // console.log(intersects[0].object.id)
```
Comment (n2oneProgrammer) [inline static/js/Game.js:233]: useless

File: static/js/Game.js [hunk 5]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+                        color: 0x00ff00, transparent: false,
+                        opacity: 1,
+                    });
+
+                    lastClickedPieceImageObj.material = new THREE.MeshBasicMaterial({
+                        color: 0x00ff00, transparent: false,
+                        opacity: 1,
+                    })
+
+                    //Znalezienie ostatnio klikniętego elementu po jego id i podmiana jego mesha jw
```
Comment (n2oneProgrammer) [inline static/js/Game.js:244]: dont comment "useless code"

File: static/js/Game.js [hunk 6]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+
+
+                    //console.log(intersects[0].object);
+
+                    //Funkcja odpowiedzialna za znalezienie elementów z piec
+                }
             }
         }
+        lastClickedPieceImageObj = intersects[0].object
+        console.log(lastClickedPieceImageObj)
```
Comment (n2oneProgrammer) [inline static/js/Game.js:259]: debug

File: static/js/Game.js [hunk 7]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+
     }
     render = () => {
         //console.log("render leci")
```
Comment (n2oneProgrammer) [inline static/js/Game.js:263]: useless

File: static/js/images.js [hunk 8]
```diff
@@ -83,7 +83,7 @@ function startImages() {
             imagesLeft.push(images[i].name)
         }
     }
-    console.log(imagesLeft)
+    //console.log(imagesLeft)
```
Comment (n2oneProgrammer) [inline static/js/images.js:86]: useless

File: static/js/images.js [hunk 9]
```diff
@@ -83,7 +83,7 @@ function startImages() {
...
 
     let XXXX = images
```
Comment (n2oneProgrammer) [inline static/js/images.js:88]: nice name of var

Comment (n2oneProgrammer) [reply static/js/images.js:88]: say everything

File: static/js/images.js [hunk 10]
```diff
@@ -105,7 +105,7 @@ function startImages() {
         // console.log(XXXX[rN].name)
 
     }
-    console.log(startingItemsArray)
+    //console.log(startingItemsArray)
```
Comment (n2oneProgrammer) [inline static/js/images.js:108]: useless

File: static/js/Game.js [hunk 11]
```diff
@@ -15,7 +15,10 @@ import { images, imagesLeft, startImages } from "../js/images.js"
 
 let zmiennaX = 4.6
```
Comment (n2oneProgrammer) [inline static/js/Game.js:16]: nice var, I know everything

**Review by n2oneProgrammer — CHANGES_REQUESTED** [review]
Hi

**Comment by jaanonim** [comment]
LGTM

File: static/js/Game.js (hunk 11 above)
Comment (jaanonim) [inline static/js/Game.js:16]: missing comma

File: static/js/Game.js [hunk 12]
```diff
@@ -15,7 +15,10 @@ import { images, imagesLeft, startImages } from "../js/images.js"
...
 let zmiennaZ = 3.6
```
Comment (jaanonim) [inline static/js/Game.js:17]: missing comma

File: static/js/Game.js [hunk 13]
```diff
@@ -15,7 +15,10 @@ import { images, imagesLeft, startImages } from "../js/images.js"
...
+
+//global helpers
 let imageCounterStrike = 0
```
Comment (jaanonim) [inline static/js/Game.js:20]: missing comma

File: static/js/Game.js [hunk 14]
```diff
@@ -15,7 +15,10 @@ import { images, imagesLeft, startImages } from "../js/images.js"
...
+let lastClickedPieceImageObj
```
Comment (jaanonim) [inline static/js/Game.js:21]: missing comma

File: static/js/Game.js [hunk 15]
```diff
@@ -114,11 +117,10 @@ class Game {
 
                 if (floor[i + 4][j + 7] == 1) {
                     let playerID = 3
```
Comment (jaanonim) [inline static/js/Game.js:119]: missing comma

File: static/js/Game.js [hunk 16]
```diff
@@ -114,11 +117,10 @@ class Game {
...
-                    console.log(j * i + i)
                     let imageOnTopName = startImages()[imageCounterStrike]
```
Comment (jaanonim) [inline static/js/Game.js:120]: missing comma

File: static/js/Game.js [hunk 17]
```diff
@@ -114,11 +117,10 @@ class Game {
...
                     let topMaterialPath = `./gfx/${imageOnTopName}.png`;
                     let pieceID = imageOnTopName
```
Comment (jaanonim) [inline static/js/Game.js:122]: missing comma

File: static/js/Game.js [hunk 18]
```diff
@@ -217,21 +219,45 @@ class Game {
         }
     }
     clickPieces = () => {
+        if (lastClickedPieceImageObj != undefined)
```
Comment (jaanonim) [inline static/js/Game.js:222]: missing comma

File: static/js/Game.js [hunk 19]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+            console.log("ostatnio kliknięty: " + lastClickedPieceImageObj.pieceID)
```
Comment (jaanonim) [inline static/js/Game.js:223]: missing comma

File: static/js/Game.js [hunk 20]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+                    // console.log(intersects[0].object.id)
+                    intersects[0].object.material = new THREE.MeshBasicMaterial({
+                        color: 0x00ff00, transparent: false,
+                        opacity: 1,
+                    });
+
+                    lastClickedPieceImageObj.material = new THREE.MeshBasicMaterial({
+                        color: 0x00ff00, transparent: false,
+                        opacity: 1,
+                    })
```
Comment (jaanonim) [inline static/js/Game.js:242]: missing comma

File: static/js/Game.js [hunk 21]
```diff
@@ -217,21 +219,45 @@ class Game {
...
+                    // lastClickedPieceImageObj.id
+
+
+                    //console.log(intersects[0].object);
+
+                    //Funkcja odpowiedzialna za znalezienie elementów z piec
+                }
             }
         }
+        lastClickedPieceImageObj = intersects[0].object
```
Comment (jaanonim) [inline static/js/Game.js:258]: missing comma

File: static/js/Game.js (hunk 6 above)
Comment (jaanonim) [inline static/js/Game.js:259]: missing comma

File: static/js/images.js [hunk 22]
```diff
@@ -83,7 +83,7 @@ function startImages() {
             imagesLeft.push(images[i].name)
```
Comment (jaanonim) [inline static/js/images.js:83]: missing comma

File: static/js/images.js [hunk 23]
```diff
@@ -105,7 +105,7 @@ function startImages() {
         // console.log(XXXX[rN].name)
 
     }
-    console.log(startingItemsArray)
+    //console.log(startingItemsArray)
     return startingItemsArray
```
Comment (jaanonim) [inline static/js/images.js:109]: missing comma

**Review by jaanonim — COMMENTED** [review]
Roo style

**Review by kdrzewiecki101 — APPROVED** [review]

**Pull request merged**